### PR TITLE
add GalaxyManga

### PR DESCRIPTION
**Pull request by MikeZeDev**

### CODE REVIEW

File: web/src/engine/websites/GalaxyManga.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        chapterData: {
+            webtoon: string[]
+        }
+    }
+}
+
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    private readonly apiurl = 'https://api.flixscans.com';
```
Comment (ronny1982) [inline web/src/engine/websites/GalaxyManga.ts:40]: May append `/api/v1/webtoon`

File: web/src/engine/websites/GalaxyManga.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/series/[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const id = new URL(url).pathname.split('-')[1];
+        const request = new FetchRequest(new URL(`/api/v1/webtoon/series/${id}`, this.apiurl).href);
```
Comment (ronny1982) [inline web/src/engine/websites/GalaxyManga.ts:56]: May use
```
FetchRequest(`${this.apiurl}/.../${id}`);
```

File: web/src/engine/websites/GalaxyManga.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        const request = new FetchRequest(new URL(`/api/v1/webtoon/homepage/latest/home?page=${page}`, this.apiurl).href);
+        const { data } = await FetchJSON<APIMangas>(request);
+        return data.map(manga => new Manga(this, provider, String(manga.id), manga.title));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const uri = new URL(`/api/v1/webtoon/chapters/${manga.Identifier}-desc`, this.apiurl);
+        const request = new FetchRequest(uri.href);
+        const data = await FetchJSON<APIChapter[]>(request);
+        return data.map(chapter => new Chapter(this, manga, chapter.id.toString(), `${chapter.name} ${chapter.title || ''}`.trim()));
```
Comment (ronny1982) [inline web/src/engine/websites/GalaxyManga.ts:80]: Does it make sense to use chapter name + title?

**Review by ronny1982 — APPROVED** [review]

File: web/src/engine/websites/GalaxyManga.ts (hunk 3 above)
Comment (MikeZeDev) [reply web/src/engine/websites/GalaxyManga.ts:80]: `name `is just the chapter number. `title `is the actual title of the chapter, when its not null. 
If possible, its better to have "40 the rabbit' returns than just '40' for chapter 40.

File: web/src/engine/websites/GalaxyManga.ts (hunk 2 above)
Comment (MikeZeDev) [reply web/src/engine/websites/GalaxyManga.ts:56]: hum, ok

File: web/src/engine/websites/GalaxyManga.ts (hunk 1 above)
Comment (MikeZeDev) [reply web/src/engine/websites/GalaxyManga.ts:40]: Not a good idea

`return data.chapter.chapterData.webtoon.map(image => new Page(this, chapter, new URL(`/storage/${image}`, this.apiurl)));`

**Pull request merged**